maestro/maestro.py

```diff
@@ -187,21 +187,27 @@ def _gather_dependencies(self, containers, forward=True):
             result = result.union(deps)
         return result
 
-    def _to_containers(self, things, expand_services):
+    def _to_containers(self, things, expand_services, container_filter):
         """Transform a list of "things", container names or service names, to
         an expended list of Container objects.
 
         Args:
             expand_services (boolean): whether to allow expanding service names
                 to their container instance names. If False and a service name
                 is encountered, the method will throw an exception.
+            container_filter (string): optional pattern to filter container names
         """
+        def do_filter(containers):
+            if not container_filter:
+                return containers
+            return [c for c in containers if fnmatch.fnmatch(container_filter, c)]
+
```

Comment:
Why not putting this logic line 218 before the final return? You'd save the nested function and the two call sites lines 207 and 210, in favor of a single place before the final `return sorted(set(result))`.

Reply:
Ack